src/components/task/MemberTaskAdminModal.tsx

```diff
@@ -208,8 +222,15 @@ const MemberTaskAdminModal: React.FC<
           })
 
           if (enabledModules.meet_service && formMeetingGateway === 'zoom') {
-            const availableZoomServiceId = await zoomServiceCheck({ overlapMeets })
-            toBeUsedServiceId = availableZoomServiceId
+            if (
+              memberTaskExecutorId !== formExecutorId &&
+              (memberTaskDueAtTime === formDueAtTime || memberTaskMeetingHours === formMeetingHours)
+            ) {
+              toBeUsedServiceId = overlapMeets.find(meet => meet.id === memberTaskMeetId)?.serviceId || null
+            } else {
+              const availableZoomServiceId = await zoomServiceCheck({ overlapMeets })
+              toBeUsedServiceId = availableZoomServiceId
+            }
```

Comment:
could move `deleteMeet` before `const { overlapMeets } = await getOverlapMeets`
then the condition line:225 to line:229 could be remove

Reply:
Thanks to review, it has been adjusted and refactored.